### DISCUSSION TITLE
Upgrade MIN and MAX

### DISCUSSION THREAD

**Discussion by Microzibert** [masterking32/MasterHamsterKombatBot#57] [Q&A] [answered]
```python
"auto_upgrade": True,
"auto_upgrade_start": 150000000,
"auto_upgrade_min": 100000000,
``` 
I set these values in the program, but instead of following the restrictions, the bot simply bought all possible cards and spent the entire balance.

 After this part of the code there are four commented lines describing the function(main.py, line 49-52), including a function that ignores min and max balance.  I would like to know how to use these functions

**Comment by cmeept** [comment, marked as answer]
Maybe you turn this option to True ?
"wait_for_best_card": False,  # Recommended to keep it True for high level accounts

**Comment by Microzibert** [comment]
I don’t think so, this setting is already set to True

**Comment by cmeept** [comment]
Change it to False and than  "auto_upgrade_start" "auto_upgrade_min" will work .
